### PR TITLE
Ux improvements

### PR DESCRIPTION
**Pull request by srivig**
Fixes:
- Company details icon bug 
- Vote button 
- handle "unknown" and "invalid" score values

### CODE REVIEW

File: mod/metrics/set/abstract/metric_answer/views.rb [hunk 1]
```diff
@@ -60,7 +60,13 @@ def default_value_link_args _args
   end
 
   def beautify value
-    card.scored? ? colorify(value) : value
+    card.scored? ? beautify_scored(value) : value
+  end
+
+  def beautify_scored value
+   no_number = Integer(value) rescue false
```
Comment (ethn) [inline mod/metrics/set/abstract/metric_answer/views.rb]: I think by the time beautify is called, the value argument may already have been altered by `humanized_value`.

File: mod/metrics/set/abstract/metric_answer/views.rb [hunk 2]
```diff
@@ -60,7 +60,13 @@ def default_value_link_args _args
   end
 
   def beautify value
-    card.scored? ? colorify(value) : value
+    card.scored? ? beautify_scored(value) : value
```
Comment (ethn) [inline mod/metrics/set/abstract/metric_answer/views.rb]: this is not your fault, but we should be using our set module pattern here.  everything that applies only to scores should be in metric_type/score.rb.

Maybe I'll take that on...

File: mod/metrics/set/all/colours.rb [hunk 3]
```diff
@@ -35,9 +35,18 @@ def mono_colour
     ]
   end
 
+  # Colours for non-numbers
+  def light_colour
+    [
+      "#cccccc",
+      "#e4e9f1"
+    ]
```
Comment (ethn) [inline mod/metrics/set/all/colours.rb]: this can be on one line.

File: mod/metrics/set/abstract/metric_answer/views.rb [hunk 4]
```diff
@@ -60,7 +60,13 @@ def default_value_link_args _args
...
+  end
+
+  def beautify_scored value
+   no_number = Integer(value) rescue false
+   value = value.to_sym == :Unknown ? "?" : "!" unless no_number
```
Comment (ethn) [inline mod/metrics/set/abstract/metric_answer/views.rb]: we shouldn't have to do this test a bunch.  

card.numeric_value? needs to handle invalid.

I would also note that it's quite confusing that `no_number` is true if it's a number and false if it's not.

I want to rearrange things a bit.  If you can get codeclimate and semaphore passing, I'll tackle the needed code improvements.

File: mod/metrics/set/all/colours.rb [hunk 5]
```diff
@@ -35,12 +35,24 @@ def mono_colour
     ]
   end
 
+  # Colours for non-numbers
+  def light_colour
```
Comment (xithan) [inline mod/metrics/set/all/colours.rb]: Everywhere else including css it's `color` not `colour`

File: mod/metrics/set/all/colours.rb [hunk 6]
```diff
@@ -35,12 +35,24 @@ def mono_colour
...
+  end
+
   def colorify value, mono=false
     return "" if value == ""
-    colour = pick_colour(value, mono)
+    colour = numeric?(value) ? pick_colour(value, mono) : light_colour[0]
     wrap_color_div(value, colour)
   end
 
+  def numeric? value
```
Comment (xithan) [inline mod/metrics/set/all/colours.rb]: There is already a `number?` methods for strings

**Review by xithan — CHANGES_REQUESTED** [review]
Isn't it a cleaner solution to use css classes (like `mono-color-gradient-1`, `multi-color-gradient-5`) instead of inline styles?

File: mod/metrics/set/all/colours.rb (hunk 6 above)
Comment (srivig) [reply mod/metrics/set/all/colours.rb]: Ah, I wasn't aware

File: mod/metrics/set/all/colours.rb (hunk 5 above)
Comment (srivig) [reply mod/metrics/set/all/colours.rb]: Shouldn't we use British English within code :D ?

**Review by ethn — APPROVED** [review]

**Review by xithan — APPROVED** [review]

**Pull request closed without merging**